### DISCUSSION TITLE
Character limit of BITO AI.

### DISCUSSION THREAD

**Discussion by Arturohgm** [gitbito/bitoai#229] [Q&A] [answered]
Hello, everyone.
I have a question.
The documentation for BITO AI states that the character limit is 140,000 with the 10X Developer plan.
My question is: What is the character limit for responses provided by BITO in a query?

**Comment by nisha-bito** [comment]
hi @Arturohgm ,

Bito uses different models and based on basic and advance query the output token size differs [4k for basic vs 8k/16k for advance].
For advance also, Bito uses different models and output token depends on model [8k or 16k].
Very soon we are releasing a feature where you can select the model from IDE before firing your query. 

Thanks,
Nisha

**Reply by Arturohgm** [reply]
Thank you for your response. I am interested in knowing how many characters BITO AI returns in the chat. I want to see this because I am developing an application with BITO for code migration, sanitization, and updating.
In the case when I pass a code file, it returns it to me incomplete. In fact, I have the 10X developer plan.

**Reply by bitoandy** [reply, marked as answer]
@Arturohgm ,

A single token can represent between 3 to 4 characters on average. So the max output can range between 24,000 to 32,000 characters (assuming 8K output token limit).
Prompts used also influence the output. Although larger input can be provided as the output has limits, one of the approaches would be to provide code blocks which are smaller so that if completely migrated to another language will fit into 8K token limits (ideally lesser than 24,000 characters to be on safer side) as there might be explanation added by AI model which will also take up the output space, which later you will need to filter out.

**Reply by Arturohgm** [reply]
Thank you so much.
Is there a link or document where that information comes?

**Reply by bitoandy** [reply]
Regarding approximate number of characters per token, checkout:
- https://docs.anthropic.com/en/docs/resources/glossary#tokens
- https://help.openai.com/en/articles/4936856-what-are-tokens-and-how-to-count-them
